### PR TITLE
Some nicer types for lodash's function helpers.

### PR DESCRIPTION
**Pull request by cqfd**
This PR adds some fancier type annotations to some of lodash's function helpers. I've added some tests as well, although they admittedly feel a bit hacky–I figured I'd start a PR and see if y'all have ideas for how to improve them.

### CODE REVIEW

File: definitions/npm/lodash_v4.x.x/flow_v0.40.x-/lodash_v4.x.x.js [hunk 1]
```diff
@@ -0,0 +1,511 @@
...
+
+    // Function
+    after<Args: Array<any>, R>(n: number, fn: (...args: Args) => R): (...args: Args) => R | void;
+    ary(func: Function, n?: number): Function;
+    before<Args: Array<any>, R>(n: number, fn: (...args: Args) => R): (...args: Args) => R | void;
+    bind(func: Function, thisArg: any, ...partials: Array<any>): Function;
+    bindKey(obj: Object, key: string, ...partials: Array<any>): Function;
+    curry(func: Function, arity?: number): Function;
+    curryRight(func: Function, arity?: number): Function;
+    debounce<F: Function>(func: F, wait?: number, options?: DebounceOptions): F;
```
Comment (cqfd) [inline definitions/npm/lodash_v4.x.x/flow_v0.40.x-/lodash_v4.x.x.js]: This bounded polymorphism lets us express that `_.debounce` returns a function of the exact same type as its input function.

File: definitions/npm/lodash_v4.x.x/flow_v0.40.x-/lodash_v4.x.x.js [hunk 2]
```diff
@@ -0,0 +1,511 @@
...
+    sortBy<T>(array: ?Array<T>, ...iteratees?: Array<Iteratee<T>>): Array<T>;
+    sortBy<T>(array: ?Array<T>, iteratees?: Array<Iteratee<T>>): Array<T>;
+    sortBy<V, T: Object>(object: T, ...iteratees?: Array<OIteratee<T>>): Array<V>;
+    sortBy<V, T: Object>(object: T, iteratees?: Array<OIteratee<T>>): Array<V>;
+
+    // Date
+    now(): number;
+
+    // Function
+    after<Args: Array<any>, R>(n: number, fn: (...args: Args) => R): (...args: Args) => R | void;
```
Comment (cqfd) [inline definitions/npm/lodash_v4.x.x/flow_v0.40.x-/lodash_v4.x.x.js]: This trick of being generic over the arguments to `fn` lets us say that `after` returns a function that takes the exact same arguments as its input, `fn`, and returns either whatever `fn` returns, or `void`.

File: definitions/npm/lodash_v4.x.x/flow_v0.38.x-v0.39.x/lodash_v4.x.x.js [hunk 3]
```diff
@@ -230,20 +230,20 @@ declare module 'lodash' {
     now(): number;
 
     // Function
-    after(n: number, fn: Function): Function;
+    after<R>(n: number, fn: (...args: Array<any>) => R): (...args: Array<any>) => R | void;
```
Comment (cqfd) [inline definitions/npm/lodash_v4.x.x/flow_v0.38.x-v0.39.x/lodash_v4.x.x.js]: The annotations for flow v.40+ are a bit fancier. As far as I can tell, all we can express here is that the function returned by `after` returns `fn`'s return type, or `void`; we can't express that its arguments should be exactly the same as `fn`'s.

**Comment by GAntoine** [comment]
Hey @cqfd, I'm liking the fancier types. `lodash` has changes quite a bit since you opened this PR. If you're still interested in seeing these changes merged, I'd encourage you to open another PR, with the fancy types added to the new definitions.

It also seems Travis didn't run the specs, so I'll be closing this PR for both of those reasons.

**Pull request closed without merging**